src/html_files/cpu_utilization.ts

```diff
@@ -1,6 +1,59 @@
 let got_cpu_util_data = false;
 let util_cpu_list: Map<string, CPUList> = new Map<string, CPUList>();
 
+let cpu_utilization_rules = {
+    data_type: "cpu_utilization",
+    pretty_name: "CPU Utilization",
+    single_run_rules: [
+        {
+            name: "User",
+            func: function (opts) {
+                let system_util = get_data_key(opts.data_type, "System");
+                let total_util: number = opts.base_run_data + system_util.get(opts.base_run);
+                if (total_util < 50) {
+                    return new Finding(
+                        `Average CPU Utilization for '${opts.base_run}' is less than 50%.`,
+                        Status.NotGood,
+                    );
+                }
+            },
+        },
+        {
+            name: "Idle",
+            func: function (opts) {
+                if (opts.base_run_data > 50) {
+                    return new Finding(`Average Idle time for '${opts.base_run}' is greater than 50%.`, Status.NotGood);
+                }
+            },
+        }
+    ],
+    per_run_rules: [
```

Comment:
There is still quite a bit of copy-paste here, could make more sense to have 1 rule, but multiple functions to implement the logic for each way to compare the run(s) to each other.  Then the rule running engine just iterates over the functions in each rule to generate the report.

Reply:
I've updated it to have a per_run, all_run and a single_run attached to one rule.